src/transforms/remap.rs

```diff
@@ -84,13 +87,24 @@ impl FunctionTransform for Remap {
 
         match result {
             Ok(_) => output.push(event),
-            Err(error) => {
+            Err(Terminate::Abort) => {
+                emit!(RemapMappingAbort {
+                    event_dropped: self.drop_on_abort,
+                });
+
+                if self.drop_on_abort {
+                    return;
+                }
+
+                output.push(event)
```

Comment:
Should this actually be pushing the original event? In other words, do we want an `abort` to abandon any changes made in the script, or should it keep any changes made up to the point `abort` was called (assuming `drop_on_abort` is false)?

If we do, I notice that the `abort` function as actually marked as infallible. Presumable, this would mean either we would have to abandon the check above and clone even if the script is infallible, or we need to mark abort as fallible.

Reply:
This is a good question! My inkling would be to have `abort` not persist any modifications.

Maybe @binarylogic has an opinion.

Reply:
My idea was it should. But I’ll defer to your decision

Reply:
If it does keep the event, then it isn't really doing much more that an `if` statement.

Reply:
`abort` says to me that the entire script should be aborted and have similar similar effects as if there were an error: namely that any mutations are discarded.

If we called it `return` on the other hand, I could see preserving the modifications.

Reply:
The term "abort" feels like it should abort the script and _not_ persist changes. Alternatively, something like "emit" (like we use in Lua) might be better served for preserving changes.